h2o-core/src/main/java/water/parser/FVecParseWriter.java

```diff
@@ -111,15 +117,20 @@ public FVecParseWriter(Vec.VectorGroup vg, int cidx, Categorical[] categoricals,
   }
   @Override public FVecParseWriter close(Futures fs){
     if( _nvs == null ) return this; // Might call close twice
+    long mem = 0;
     for(int i=0; i < _nvs.length; i++) {
       _nvs[i].close(_cidx, fs);
+      mem += _nvs[i].chk2().byteSize();
```

Comment:
Why do we need the "byteSize" method on NewChunk? At this point, NewChunk is already compressed (close method was called) and we know the exact size in memory.

I don't think I understand the concept.

Reply:
at this point byteSize is 0, we need byteSize to compute size of newChunk in memory before it gets compressed, after newchunk is compressed we use the size of compressed chunk

Reply:
okay, I see - you are monitoring even the chunks that are just being written - shooting for the stars :)

now I am closer to the understanding

Reply:
interesting point, counting just finished chunks would certainly simplify things, but during parsing in-progress chunks take significant amount of memory so I would like to keep this